test/unit/serializer.cpp

```diff
@@ -256,32 +263,35 @@ struct serializer_test
         check_src(
             "HTTP/1.1 200 OK\r\n"
             "Server: test\r\n"
-            "Content-Length: 5\r\n"
+            "Content-Length: 2048\r\n"
             "\r\n",
-            test_source{},
-            //--------------------------
-            "HTTP/1.1 200 OK\r\n"
-            "Server: test\r\n"
-            "Content-Length: 5\r\n"
-            "\r\n"
-            "12345");
+            test_source{std::string(2048, 'A')},
+            [](core::string_view s){
+                BOOST_TEST(s ==
+                    "HTTP/1.1 200 OK\r\n"
+                    "Server: test\r\n"
+                    "Content-Length: 2048\r\n"
+                    "\r\n" +
+                    std::string(2048, 'A'));
+            });
 
         // source chunked
         check_src(
             "HTTP/1.1 200 OK\r\n"
             "Server: test\r\n"
             "Transfer-Encoding: chunked\r\n"
             "\r\n",
-            test_source{},
-            //--------------------------
-            "HTTP/1.1 200 OK\r\n"
-            "Server: test\r\n"
-            "Transfer-Encoding: chunked\r\n"
-            "\r\n"
-            "0000000000000005\r\n"
-            "12345"
-            "\r\n"
-            "0\r\n\r\n");
+            test_source{std::string(1024, 'A')},
+            [](core::string_view s){
+                core::string_view expected_header = 
+                    "HTTP/1.1 200 OK\r\n"
+                    "Server: test\r\n"
+                    "Transfer-Encoding: chunked\r\n"
+                    "\r\n";
+                BOOST_TEST(s.starts_with(expected_header));
+                s.remove_prefix(expected_header.size());
+                BOOST_TEST(std::count(s.begin(), s.end(), 'A') == 1024);
+            });
```

Comment:
A better alternative might be to create a basic chunk decoder and assert the expected number of chunks and their content.
I'm looking for ideas to see if there is a simpler way of doing it.